pyrefly/lib/test/operators.rs

```diff
@@ -581,6 +614,6 @@ class A:
 class B:
     def __radd__(self, other) -> Self:
         return self
-assert_type(A() + B(), A)  # E: `A.__add__` is deprecated
+assert_type(A() + B(), A | B)  # E: `A.__add__` is deprecated
```

Comment:
does this A | B imply we choose A | B in [this ](https://github.com/facebook/pyrefly/issues/1176#issuecomment-3432529675)example?


is it worth adding a test for that, and should it be just A?

Reply:
The current behavior for these reflected operators is:
- Try the "forward" operator, if that works, use the return type from that
- If the forward operator fails, try the reflected operator, using it's return type if it succeeds
- If both the forward and reflected operators fail, error and use the return type from the forward operator

is this level of strictness practical? might be worth a discussion